src/build/arg/mod.rs

```diff
@@ -2299,6 +2300,111 @@ impl<'help> Arg<'help> {
         self
     }
 
+
+    /// Specifies a value for the argument when the argument is supplied and a value is required
+    /// but the value is *not* specified at runtime.
+    ///
+    /// This configuration option is often used to give the user a shortcut and allow them to
+    /// efficiently specify an option argument without requiring an explicitly value. The `--color`
+    /// argument is a common example. By, supplying an default, such as `default_missing_value("always")`,
+    /// the user can quickly just add `--color` to the command line to produce the desired color output.
+    ///
+    /// **NOTE:** using this configuration option requires the use of the `.min_values(0)` and the
+    /// `.require_equals(true)` configuration option. These are required in order to unambiguously
+    /// determine what, if any, value was supplied for the argument.
+    ///
+    /// # Examples
+    ///
+    /// Here is an implementation of the common POSIX style `--color` argument.
+    ///
+    /// ```rust
+    /// # use clap::{App, Arg};
+    /// let a = App::new("prog")
+    ///         .arg(Arg::new("color")
+    ///             .value_name("WHEN")
+    ///             .possible_values(&["always", "auto", "never"])
+    ///             .default_value("auto")
+    ///             .overrides_with("color")
+    ///             .min_values(0)
+    ///             .require_equals(true)
+    ///             .default_missing_value("always")
+    ///             .about(r#"Specify WHEN to colorize output.
+    ///                 [WHEN] == ( *always* | auto | never ) ; "--color=auto" is assumed if the option is never specified
```

Comment:
This message is too long. Won't the first line be enough?

Reply:
First line is fine, but when I use this, I'll use the whole line. It attempts to explain that the option has a default (`auto`) and a different default (default_missing_value) for an argument specified on the command line without a value (eg, `--color` => `--color=always`). It's the best construction that I've come up with at the moment.
I wanted to give the user a prompt to use something similar.